k8s/core/configmap/configmap_lock_v1.go

```diff
@@ -44,7 +44,12 @@ func (c *configMap) Unlock() error {
 	fn := "Unlock"
 	// Get the existing ConfigMap
 	c.kLockV1.Lock()
-	defer c.kLockV1.Unlock()
+	defer func() {
+		// Check if the mutex is locked before unlocking
+		if !c.kLockV1.unlocked {
+			c.kLockV1.Unlock()
+		}
+	}()
```

Comment:
unlocked is set to true always in line 57. We will not unlock this every in that case. I don't think that should be the fix here. We need to see where c.KLockV1's Unlock is happening more than once

Reply:
The only thing I can think about is that klockv1 changed after we did the lock because of which it is unlocked state. Is it possible that klockv1 during the life of the cm.

Reply:
The other possibility is that some other path did an unlock unnecessarily. 
I had a chat with Rushi. He said he is able to reproduce the issue. I have suggested to trace all locks and unlocks and find an unexpected unlock that might have occured.